module/zfs/arc.c

```diff
@@ -6254,11 +6255,10 @@ arc_tuning_update(void)
 		arc_c_max = zfs_arc_max;
 		arc_c = arc_c_max;
 		arc_p = (arc_c >> 1);
-		/* Valid range of arc_meta_limit: arc_meta_min - arc_c_max */
-		percent = MIN(zfs_arc_meta_limit_percent, 100);
-		arc_meta_limit = MAX(arc_meta_min, (percent * arc_c_max) / 100);
-		percent = MIN(zfs_arc_dnode_limit_percent, 100);
-		arc_dnode_limit = (percent * arc_meta_limit) / 100;
+		if (arc_meta_limit > arc_c_max)
```

Comment:
Good question! I didn't really consider it, but, with the help of a little bit of post rationalising, to me it's clearer what the code is doing with the explicit test, i.e. "if the value is greater than the limit, set it to the limit" rather than "set the value to the minimum of the value itself and the limit". To my brain the first of those prose expansions is much (much!) clearer than the second, and that prose is a form of what goes on in my head when I read the code.

That said, you're correct that it makes that block of code somewhat inconsistent.

If anyone cares ( @behlendorf , @dweeezil ?) I can make it consistent - either by changing my code to MIN/MAX, or changing the MIN/MAX to explicit code (except the `MIN(zfs_arc_dnode_limit_percent, 100)` instances which "scan" properly in my head, particularly given the added complexity of the ternary operator).